src/main/java/org/isf/medicalstockward/rest/MedicalStockWardController.java

```diff
@@ -0,0 +1,242 @@
+		if(mappedMovs.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(mappedMovs);
+		} else {
+			return ResponseEntity.ok(mappedMovs);
+		}
+	}
+	
+	/**
+	 * Gets all the movement ward with the specified criteria.
+	 * @param idwardTo the target ward id.
+	 * @param dateFrom the lower bound for the movement date range.
+	 * @param dateTo the upper bound for the movement date range.
+	 * @return all the retrieved movements.
+	 * @throws OHServiceException 
+	 */
+	@GetMapping(value = "/medicalstockward/movements/to/{target_ward_code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<MovementWardDTO>> getWardMovementsToWard(
+			@PathVariable("target_ward_code") String idwardTo, 
+			@RequestParam("from") Date dateFrom, 
+			@RequestParam("to") Date dateTo) throws OHServiceException {
+		GregorianCalendar dateFrom_ = null;
+		if(dateFrom != null) {
+			dateFrom_ = new GregorianCalendar();
+			dateFrom_.setTime(dateFrom);
+		}
+		
+		GregorianCalendar dateTo_ = null;
+		if(dateTo != null) {
+			dateTo_ = new GregorianCalendar();
+			dateTo_.setTime(dateTo);
+		}
+		
+		List<MovementWard> movs = movWardBrowserManager.getWardMovementsToWard(idwardTo, dateFrom_, dateTo_);
+		List<MovementWardDTO> mappedMovs = movementWardMapper.map2DTOList(movs);
+		if(mappedMovs.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(mappedMovs);
+		} else {
+			return ResponseEntity.ok(mappedMovs);
+		}
+	}
+	
+	/**
+	 * Persists the specified movement.
+	 * @param newMovementDTO the movement to persist.
+	 * @return <code>true</code> if the movement has been persisted, <code>false</code> otherwise.
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/medicalstockward/movements", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newMovementWard(@Valid @RequestBody MovementWardDTO newMovementDTO) throws OHServiceException {
+		MovementWard newMovement = movementWardMapper.map2Model(newMovementDTO);
+		boolean isPersisted = movWardBrowserManager.newMovementWard(newMovement);
+		if (!isPersisted) {
+            throw new OHAPIException(new OHExceptionMessage(null, "Movement ward is not created!", OHSeverityLevel.ERROR));
+        }
+        return ResponseEntity.status(HttpStatus.CREATED).body(null);
+	}
+	
+	/**
+	 * Persists the specified movements.
+	 * @param newMovementDTOs the movements to persist.
+	 * @return <code>true</code> if the movements have been persisted, <code>false</code> otherwise.
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/medicalstockward/movements/all", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newMovementWard(@Valid @RequestBody List<MovementWardDTO> newMovementDTOs) throws OHServiceException {
+		ArrayList<MovementWard> newMovements = new ArrayList<MovementWard>();
+		movementWardMapper.map2ModelList(newMovementDTOs).forEach(mov -> {
+			newMovements.add(mov);
+		});
```

Comment:
We probably avoid foreach iteration assigning movementWardMapper.map2ModelList(newMovementDTOs) to newMovements?